File: Hash Map/hashmap.js
class HashMap {
    constructor(capacity) {
        this.hashmap = {};
        for (var i = 0; i < capacity; i++){
            this.hashmap[i] = [];
        }
        this.capacity = capacity;
    }

    hashfunction(key) {
        return key % this.capacity;
    }

    insertion(key, val) {
        var index = this.hashfunction(key);

        this.hashmap[index].push([key, val]);
    }

    lookup(key) {
        var index = this.hashfunction(key);
        for (var i = 0; i < this.hashmap[index].length; i++){
            if (this.hashmap[index][i][0] == key) {
                return this.hashmap[index][i][1]
            }
        }
        return "Could not find key"
    }

    deletion(key) {
        var index = this.hashfunction(key);
        var found = false
        for (var i = 0; i < this.hashmap[index].length; i++) {
            if (this.hashmap[index][i][0] == key) {
                this.hashmap[index].splice(i, 1)
                found = true
            }
        }
        if (!found) {
            console.log("Could not find key")   
        }
    }
}

File: Hash Map/hashmap.py
class HashMap():
    def __init__(self, capacity):
        self.hashmap = {x:[] for x in range(capacity)}
        self.capacity = capacity

    def hashfunction(self, x):
        return x % self.capacity

    def insertion(self, x, val):
        index = self.hashfunction(x)

        # Collision resolution using seperate chaining
        self.hashmap[index].append((x, val))

        
    def lookup(self, key):
        index = self.hashfunction(key)
        for k, val in self.hashmap[index]:
            if k == key:
                return val

        return "Could not find key"

    def deletion(self, key):
        index = self.hashfunction(key)
        found = False
        for k, val in self.hashmap[index]:
            if k == key:
                self.hashmap[index].remove((k, val))
                found = True
                break

        if not found:
            print("Could not find key")
